pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -1283,30 +1303,41 @@ func (b *MacvtapBindMechanism) decorateConfig() error {
 	return nil
 }
 
-func (b *MacvtapBindMechanism) loadCachedInterface(pid, name string) (bool, error) {
-	ifaceConfig, err := b.storeFactory.CacheForPID(pid).Read(name)
+func (b *MacvtapBindMechanism) getPIDString() string {
+	if b.launcherPID != nil {
+		return fmt.Sprintf("%d", *b.launcherPID)
+	}
+	return "self"
+}
+
+func (b *MacvtapBindMechanism) loadCachedInterface() error {
+	ifaceConfig, err := b.storeFactory.CacheForPID(getPIDString(b.launcherPID)).Read(b.iface.Name)
 	if os.IsNotExist(err) {
-		return false, nil
+		return nil
 	}
 
 	if err != nil {
-		return false, err
+		return err
 	}
 
 	b.virtIface = ifaceConfig
-	return true, nil
+	return nil
 }
 
-func (b *MacvtapBindMechanism) setCachedInterface(pid, name string) error {
-	err := b.storeFactory.CacheForPID(pid).Write(name, b.virtIface)
-	return err
+func (b *MacvtapBindMechanism) hasCachedInterface() bool {
+	return b.virtIface != nil
 }
 
-func (b *MacvtapBindMechanism) loadCachedVIF(_, _ string) (bool, error) {
-	return true, nil
+func (b *MacvtapBindMechanism) setCachedInterface() error {
+	pidStr := fmt.Sprintf("%d", *b.launcherPID)
```

Comment:
Can we use https://github.com/kubevirt/kubevirt/pull/5059/files#diff-030aa8b7ff9bf4a511d1d5641c1dec04205a944137672be3a691f0b1b4068753R548 ?

Reply:
I don't know ... That one should be used when you read. Here, I'd like the compiler to blow up on my face when I have missed to set the PID in the structure.

Reply:
This is going to be runtime error not a compiler error in fact this is going to be a panic not a controlled error.

Reply:
Duh, right. Never reply when hungry.

Reply:
Protected against that, and used the helper func.